ui/core/src/react/v1.tsx

```diff
@@ -0,0 +1,216 @@
+import React, {
+  useEffect,
+  createContext,
+  useState,
+  useContext,
+  useMemo,
+} from "react";
+import { Store } from "../store";
+import { effect, stop } from "@vue/reactivity";
+import { isAmount, isAssetAmount } from "../entities";
+import { AccountPool } from "../store/pools";
+import { Api } from "../api";
+import { Services } from "../services";
+import { setupSifchainApi } from "../setupSifchainApi";
+
+type ExtractorFn<T> = (
+  setter: React.Dispatch<React.SetStateAction<T>>,
+  store: Store,
+) => void;
+
+/**
+ * Abstraction to extract specific state from the store
+ * @param name the hook name
+ * @param extractor a function to convert between the native store and hook output
+ * @returns data output type
+ */
+function useStateHook<T>(name: string, extractor: ExtractorFn<T>, init: T) {
+  const [state, setState] = useState<T>(init);
+  const ctx = useContext(SifchainContext);
+  if (!ctx) {
+    throw new Error(name + " must be used within a SifchainProvider");
+  }
+
+  useEffect(() => {
+    const { store } = ctx;
+    const ef = effect(() => {
+      extractor(setState, store);
+    });
+
+    return () => stop(ef);
+  }, [ctx.store]);
```

Comment:
With the way the API is formatted, extractor actually changes every render (new one passed in). It should be in this dependency array.

This whole `useStateHook` would make more sense as a hook factory, since the dependencies never change.

```js
function createStateHook (name, extractor, init) {
  return function useStateHook () {
    // ... do everything
  }
}

export const usePoolsState = createStateHook(
  "usePoolsState",
  store => clone(store.pools),
  {},
);

Reply:
So one issue with the factory stuff is that it prevents us from easily providing arguments to the extractor. We could create overrides but we don't get as dynamic control over our arg list of the hooks we are creating. So instead I figured it best to just solve the creation on render issue by extracting out the args to an object that is pulled into the closure.

Reply:
Ok this is starting to look better.